### PR TITLE
fix: fetch more data error

### PR DESCRIPTION
**Pull request by ardasnturk**
- updated the `query` in the `fetchMoreData` function with `genericItems` because the data from the server is in the `genericItems` object
- added `_filter` by `categories` to `count` value because we can't list vouchers without `categories`
- added `GET_VOUCHERS` query to `FETCH_MORE_QUERIES` to run voucher query with `getFetchMoreQuery`

SVAK-35

### CODE REVIEW

File: src/screens/Voucher/VoucherIndexScreen.tsx [hunk 1]
```diff
@@ -118,7 +126,7 @@ export const VoucherIndexScreen = ({ navigation, route }: StackScreenProps<any>)
     return <LoadingSpinner loading />;
   }
 
-  const count = listItems?.length;
+  const count = _filter(listItems, ({ categories }) => categories.length > 0).length;
```
Comment (donni106) [inline src/screens/Voucher/VoucherIndexScreen.tsx:129]: what about `listItems.filter({ categories } => !!categories.length).length`? i do not think that we need lodash here and on the other side we often not check length with > 0 but with boolean !!

Comment (ardasnturk) [reply src/screens/Voucher/VoucherIndexScreen.tsx:129]: There are also two different usages in the code. Both do the same thing. loadsh has no extra advantage. I can change the code. 

Yes you are right, I will update the category control.

Comment (ardasnturk) [reply src/screens/Voucher/VoucherIndexScreen.tsx]: https://github.com/smart-village-solutions/smart-village-app-app/commit/7d0a8ceffce23adcff5deea4e69324d54058c226

File: src/screens/Voucher/VoucherIndexScreen.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { StackScreenProps } from '@react-navigation/stack';
 import _uniqBy from 'lodash/uniqBy';
+import _filter from 'lodash/filter';
```
Comment (donni106) [inline src/screens/Voucher/VoucherIndexScreen.tsx:3]: can be removed now, right?

Comment (ardasnturk) [reply src/screens/Voucher/VoucherIndexScreen.tsx]: yeah! 👍 😄

**Review by donni106 — APPROVED** [review]

**Pull request merged**